packages/base/src/mainview/mainView.tsx

```diff
@@ -244,16 +245,41 @@ export class MainView extends React.Component<IProps, IStates> {
   private async _loadShapefileAsGeoJSON(
     url: string
   ): Promise<GeoJSON.FeatureCollection | GeoJSON.FeatureCollection[]> {
+    // First trying a direct fecth
+    try {
+      const response = await fetch(url);
+      const arrayBuffer = await response.arrayBuffer();
+      const geojson = await shp(arrayBuffer);
+
+      return geojson;
+    } catch (error) {
+      console.warn(`Cannot download file from ${url}: ${error}`);
+    }
+
+    // Trying through our proxy server
     try {
       const response = await fetch(`/jupytergis_core/proxy?url=${url}`);
       const arrayBuffer = await response.arrayBuffer();
       const geojson = await shp(arrayBuffer);
 
       return geojson;
     } catch (error) {
-      console.error('Error loading shapefile:', error);
-      throw error;
+      console.warn('Cannot communicate with the JupyterGIS proxy server', error);
     }
+
+    // Trying through an external proxy server
+    try {
+      const response = await fetch(`https://corsproxy.io/?url=${url}`);
```

Comment:
There are other CORS anywhere servers here and there, can you try to find one that you have access to?

Reply:
No free CORS server is reliable since it needs to eat all the bandwidth needed to download and serve your files. A safer bet is to make it configurable from the UI

Reply:
Indeed, making it configurable through a setting would be nice 👍🏽